agent/acl.go

```diff
@@ -155,7 +149,7 @@ func (m *aclManager) lookupACL(a *Agent, id string) (acl.ACL, error) {
 		id = anonymousToken
 	} else if acl.RootACL(id) != nil {
 		return nil, errors.New(rootDenied)
-	} else if m.master != nil && id == a.config.ACLAgentMasterToken {
+	} else if a.tokens.IsAgentMasterToken(id) {
```

Comment:
No - the token store is created as part of constructing an agent, so it will be there even if ACLs aren't enabled.